File: tests/aster/x/pl/cross_join_triple.pl
:- initialization(main).
:- style_check(-singleton).
main :-
        Nums = [1, 2],
            Letters = ['A', 'B'],
            Bools = [true, false],
            Combos = [_{b: true, l: 'A', n: 1}, _{b: false, l: 'A', n: 1}, _{b: true, l: 'B', n: 1}, _{n: 1, b: false, l: 'B'}, _{b: true, l: 'A', n: 2}, _{b: false, l: 'A', n: 2}, _{b: true, l: 'B', n: 2}, _{b: false, l: 'B', n: 2}],
            writeln('--- Cross Join of three lists ---'),
            C = _{l: 'A', n: 1, b: true},
            writeln('1 A true'),
            C1 = _{l: 'A', n: 1, b: false},
            writeln('1 A false'),
            C2 = _{b: true, l: 'B', n: 1},
            writeln('1 B true'),
            C3 = _{b: false, l: 'B', n: 1},
            writeln('1 B false'),
            C4 = _{b: true, l: 'A', n: 2},
            writeln('2 A true'),
            C5 = _{l: 'A', n: 2, b: false},
            writeln('2 A false'),
            C6 = _{b: true, l: 'B', n: 2},
            writeln('2 B true'),
            C7 = _{b: false, l: 'B', n: 2},
        writeln('2 B false').

File: archived/tests/transpiler/x/pas/cross_join_triple.pas
{$mode objfpc}
program Main;
type Anon5 = record
  n: integer;
  l: string;
  b: boolean;
end;
var
  nums: array of integer;
  letters: array of string;
  bools: array of boolean;
  combos: array of Anon5;
  c: integer;
  n: integer;
  l: string;
  b: boolean;
begin
  nums := [1, 2];
  letters := ['A', 'B'];
  bools := [true, false];
  combos := [];
  for n in nums do begin
  for l in letters do begin
  for b in bools do begin
  combos := concat(combos, [(n: n; l: l; b: b)]);
end;
end;
end;
  writeln('--- Cross Join of three lists ---');
  for c in combos do begin
  writeln(c.n, ' ', c.l, ' ', c.b);
end;
end.
